src/NewTwitchApi/Webhooks/WebhooksSubscriptionApi.php

```diff
@@ -22,7 +22,7 @@ public function __construct(string $clientId, string $secret, Client $guzzleClie
         $this->guzzleClient = $guzzleClient ?? new HelixGuzzleClient($clientId);
     }
 
-    public function subscribeToStream(string $twitchId, string $bearer, string $callback, int $leaseSeconds = 0): void
+    public function subscribeToStream(string $twitchId, string $bearer = null, string $callback, int $leaseSeconds = 0): void
```

Comment:
I'm ok making bearer null by default, given your explanation. However, I'm not a fan of having optional argument between required arguments. I realize this is largely my fault for initially writing the code this way, due to my lack of understanding that bearer isn't always required. However, is there a better way to handle this? I suppose `bearer` and `callback` could be swapped in the arguments, but that will be a pretty major (and annoying) break in backward compatibility. Alternatively, `bearer` could be `?string`, but then you'd have to pass in null every time you didn't want to use one, which is no good, either. Thoughts?

Reply:
I wasn't aware of nullable types in PHP, if I did I would have went for that. But I do agree that nullable parameters in the middle aren't ideal, on the other hand I thought that backwards compatibility must be kept. Personally, swapping the parameters in my projects would be a trivial change, but I didn't think you would consider breaking compatibility. Then again this API version is fairly new and probably is going to stick around for a while, so making the change now might save some annoyance later.

Reply:
I agree. I think swapping them now, bumping the version to a breaking change point-release which mentions the change in the release notes, would save time and trouble down the road. Hopefully people have their versions locked in composer.json and read the release notes before upgrading. If not, there's not a lot we can do about that.